src/Mvc/Mvc.ViewFeatures/src/RazorComponents/InvokedRenderModes.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+internal sealed class InvokedRenderModes
+{
+    public InvokedRenderModes(Mode mode)
+    {
+        Value = mode;
+    }
+
+    public Mode Value { get; set; }
+
+    internal enum Mode
+    {
+        None,
+        Server,
+        WebAssembly,
+        ServerAndWebAssembly
```

Comment:
What is ServerAndWebAssembly?

Reply:
Well spotted 😄

Reply:
Ah, it's something that the component renderer uses to track the render mode for persisted data. It needs to detect if you did RenderMode.Server and RenderMode.WebAssembly as it can't protect the data in that case

Reply:
Are the names a bit misleading then? Seems like we'd be less likely to get confused if the names were the same as in https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.ViewFeatures/src/RenderMode.cs

Reply:
I think it deserves adding a comment, but once you understand what's tracking the names are as explicit as they can be

Reply:
All I did was move the type 🤷🏽. I'll see if I can reverse engineer their doc comments from their usage.